common/src/services/passwordReprompt.service.ts

```diff
@@ -16,13 +16,7 @@ export class PasswordRepromptService implements PasswordRepromptServiceAbstracti
 
     async showPasswordPrompt() {
         const passwordValidator = async (value: string) => {
-            const keyHash = await this.cryptoService.hashPassword(value, null, HashPurpose.LocalAuthorization);
-            const storedKeyHash = await this.cryptoService.getKeyHash();
-
-            if (storedKeyHash == null || keyHash == null || storedKeyHash !== keyHash) {
-                return false;
-            }
-            return true;
+            return await this.cryptoService.compareAndUpdateKeyHash(value, null);
```

Comment:
async/await no longer necessary here. just return `compareAndUpdateKeyHash` unawaited.

Reply:
Is this because `showPasswordDialog` already `await`s `passwordValidator` and will therefore `await` the promise it returns?

Should the same change be made in mobile? https://github.com/bitwarden/mobile/pull/1423/files#diff-e6b6724aaae6e1c5614946a04739af5d574b83805a6827452e7b0695c423f4cdR32